### ISSUE TITLE
Abnormal when generating multiple [SEG] tokens for multiple instances training

### ISSUE THREAD

**Issue by xpwindells** [QuentinFitteRey/VLMSAM#5]
I have established a training set and a validation set for multi-instance segmentation. I used the official lisa++ for training, and the accuracy of the validation set has improved normally. However, when using this code  based qwen2.5vl library for training, the improvement of the validation set is very small. Did the author encounter any issues where the training process was abnormal when generating multiple [SEG] tokens for multiple instances?

**Comment by connorye** [comment]
Hello,May I know  what model are you training with the original code of LISA++,llava-v1-llama2 or llava-v1.5 or finetune on lisa-7b-plus?

**Comment by xpwindells** [comment]
> Hello,May I know what model are you training with the original code of LISA++,llava-v1-llama2 or llava-v1.5 or finetune on lisa-7b-plus?

I am using LISA - 7b-plus that has been re-trained.

**Comment by QuentinFitteRey** [comment]
So I'm not sure why you have those results, the data loading my be wrong in the training. However, did you try to do a training on top of your first model where you focus only on multi instance. Like 100 steps to increase the performance. Or augment the data distribution towards multi instance! Please let me know

**Comment by xpwindells** [comment]
Okay,this is my config:
parser.add_argument("--dataset", default="sem_seg||refer_seg||vqa||reason_seg||cot", type=str)
parser.add_argument("--sample_rates", default="8,3,3,3,8", type=str)


Multiple instance loading example:
['<|im_start|>system\nYou are a helpful assistant.<|im_end|>\n<|im_start|>user\n<|vision_start|><|image_pad|><|vision_end>\nWhere is the object being used to transport a child while allowing them to sit comfortably? Please output segmentation mask.Generate instance segmentation masks. Each countable object within the same category should have a separate mask.Ignore uncountable categories for individual masking.<|im_end|>\n<|im_start|>assistant\n[SEG] and [SEG].<|im_end|>']

['<|im_start|>system\nYou are a helpful assistant.<|im_end|>\n<|im_start|>user\n<|vision_start|><|image_pad|><|vision_end>\nShow the item on display that is typically used for serving food. Please output segmentation mask.Generate instance segmentation masks. Each countable object within the same category should have a separate mask.Ignore uncountable categories for individual masking.<|im_end|>\n<|im_start|>assistant\n[SEG], [SEG], [SEG], [SEG], [SEG], [SEG], [SEG], [SEG], and [SEG].<|im_end|>']


If the validation set is "reasonseg", the IoU value will increase to 0.60. If the validation set is "instance reasonseg", the IoU value reaches a maximum of 0.10 and performs poorly on my own inference samples, for example:

<img width="754" height="273" alt="Image" src="https://github.com/user-attachments/assets/e5c19be6-2be9-4911-a69e-0588af4e4712" />


if you have used the qwenvl2.5 framework to reproduce the results of multi-instance segmentation?

**Comment by xpwindells** [comment]
vlmsam.py
model_forward:
  seg_token_mask = input_ids[:, 1:] == self.seg_token_idx
        seg_token_mask = torch.cat(
            [
                seg_token_mask,
                torch.zeros((seg_token_mask.shape[0], 1)).bool().cuda(),
            ],
            dim=1,
        )
multiple [SEG] instance val dataset GIOU:0.22
Perhaps there was an error when the Qwen model extracted the hidden layer features of the tokens?

**Comment by c1ircle** [comment]
> vlmsam.py model_forward： seg_token_mask = input_ids[：， 1：] == self.seg_token_idx seg_token_mask = torch.cat（ [ seg_token_mask， torch.zeros（（seg_token_mask.shape[0]， 1））.bool（）.cuda（）， ]， dim=1， ） multiple [SEG] 实例 val 数据集 GIOU：0.22 也许Qwen模型提取代币隐藏层特征时出现了错误？

Hello! I was wondering if you managed to resolve this issue? Regarding the SEG token extraction, should we use input_ids[:, 1:] == self.seg_token_idx or input_ids[:, :] == self.seg_token_idx?
